lib/puppet_x/puppetlabs/dsc_type_helpers.rb

```diff
@@ -58,6 +58,10 @@ def self.validate_MSFT_Credential(name, value)
         unless extraneous.empty?
           fail "#{name} includes invalid keys: #{extraneous.join(',')}"
         end
+
+        if value.values.any?{|v| v.nil? || v.length == 0}
```

Comment:
Since we were already iterating over user and password a few lines above, why not just adjust the logic there rather than introduce new logic?

https://github.com/jpogran/puppetlabs-dsc/blob/20461a78cf95e64588a6eaa152824c798961ff59/lib/puppet_x/puppetlabs/dsc_type_helpers.rb#L44-L48